### DISCUSSION TITLE
2023/02/15/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%951-%E4%B8%A4%E6%95%B0%E4%B9%8B%E5%92%8C/

### DISCUSSION THREAD

**Discussion by giscus** [Zhuanjier/blog#11] [Announcements]
# 2023/02/15/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%951-%E4%B8%A4%E6%95%B0%E4%B9%8B%E5%92%8C/

两数之和原题传送门 给定一个整数数组 nums 和一个整数目标值 target，请你在该数组中找出和为目标值 target  的那两个整数，并返回它们的数组下标。 你可以假设每种输入只会对应一个答案。但是，数组中同一个元素在答案里不能重复出现。 你可以按任意顺序返回答案。 示例示例1输入：nums = [2,7,11,15], target = 9输出：[0,1]解释：因为 nums[0] + n

https://blog.zhuanjie.ltd/2023/02/15/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%951-%E4%B8%A4%E6%95%B0%E4%B9%8B%E5%92%8C/

**Comment by RandallAnjie** [comment]
Map遍历效率比较：
1）map的key采用简单形式和复杂形式时，查找的效率是不同的，简单的key值效率更高
2）当数据量大的时候，采用entrySet遍历key+value的效率要高于keySet
3）当我们只需要取得value值时，采用values来遍历效率更高
